src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -357,7 +360,7 @@ public ResponseEntity<byte[]> getLecturePdfAttachmentsMerged(@PathVariable Long
 
         List<String> attachmentLinks = lectureAttachments.stream()
                 .filter(unit -> authCheckService.isAllowedToSeeLectureUnit(unit, user) && "pdf".equals(StringUtils.substringAfterLast(unit.getAttachment().getLink(), ".")))
-                .map(unit -> FilePathService.getAttachmentUnitFilePath()
+                .peek(unit -> lectureUnitService.setLectureUnitCompletion(unit, user, true)).map(unit -> FilePathService.getAttachmentUnitFilePath()
```

Comment:
This looks like we create on additional DB call per lecture unit. Would it be a good idea to set these completion states in a single DB call for all units at once?